app/models/manageiq/providers/network_manager.rb

```diff
@@ -64,15 +64,27 @@ class << model_name
     virtual_total :total_miq_templates, :miq_templates
     virtual_total :total_vms_and_templates, :vms_and_templates
 
-    # Relationships delegated to parent manager
-    virtual_delegate :orchestration_stacks,
-                     :orchestration_stacks_resources,
-                     :direct_orchestration_stacks,
-                     :resource_groups,
-                     :key_pairs,
-                     :to        => :parent_manager,
-                     :allow_nil => true,
-                     :default   => []
+    # Relationships delegated to parent manager.
+    # wanted to use delegate, but want it to return [] when parent_manager.nil?
+    def orchestration_stacks
+      parent_manage.orchestration_stacks || []
```

Comment:
```suggestion
      parent_manager.orchestration_stacks || []
```

Reply:
Does `default` not handle a nil relationship?  Should it?

Reply:
@kbrock are you saying that `virtual_has_many :through` behaves exactly as it previously did regarding nils/defaults with

```ruby
virtual_delegate
...
                     :to        => :parent_manager,
                     :allow_nil => true,
                     :default   => []
```

Do you still have concerns about changes in behavior with this change?

Reply:
So `virtual_has_many` was augmented as so:

```
define_delegate(name, source || name, :to => through, :allow_nil => true, :default => []) if through
```

so it created a method named `name` that linked to the `through` (e.g.: `parent_manager`), with a name of the `name`, unless the user overrode (not the case here). It defaults to `[]`, and it does `allow_nil`.